src/main/java/mate/academy/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,38 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.MovieDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    private final SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public Movie add(Movie movie) {
+        Session session = sessionFactory.openSession();
+        Transaction transaction = session.beginTransaction();
+        session.save(movie);
+        transaction.commit();
+        session.close();
+        return movie;
+    }
+
+    @Override
+    public Optional<Movie> get(Long id) {
+        Session session = sessionFactory.openSession();
+        Movie movie = session.get(Movie.class, id);
+        session.close();
+        try {
+            return Optional.of(movie);
+        } catch (Exception e) {
+            throw new DataProcessingException("Could not get Movie by Id: " + id, e);
+        }
```

Comment:
```suggestion
   try (Session session = HibernateUtil.getSessionFactory().openSession()) {
            return Optional.ofNullable(session.get(Movie.class, id));
        } catch (Exception e) {
            throw new DataProcessingException("Can't get movie by id: " + id, e);
        }```

Reply:
What happens if you have null here? You don't handle this situation. Your service method will throw an exception